ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandStore.java

```diff
@@ -245,8 +251,18 @@ public void ensureConsumedPast(final long seqNum, final Duration timeout)
     return result;
   }
 
+  private long getSequenceNumberToWaitFor(final long seqNum) {
+    return seqNum == CommandStoreUtil.LAST_SEQUENCE_NUMBER ? lastCommandSequenceNumber : seqNum;
```

Comment:
Instead of tracking `lastCommandSequenceNumber` and using that value here when `seqNum == LAST_SEQUENCE_NUMBER`, ideally we would call `commandConsumer.endOffsets` here directly, to truly get the current last sequence number in the command queue. However, `commandConsumer` does not support multi-threading and calls to this function are made by server threads issuing requests, in contrast to all other calls to `commandConsumer` which are made by the single command runner thread. Thus, tracking `lastCommandSequenceNumber` and using it here was the best workaround I could think of. Open to other suggestions!

Reply:
Ask discussed in the meeting, this won't work as there is a race condition between the threads, i.e. its possible to receive a request that wasn't to be executed _after_ all currently queued requests. So the code checks `lastCommandSequenceNumber` upon receiving the request, but it hasn't been updated since another command was posted, so its actually out of date... not good!

As discussed, use a separate thread with its own consumer. Use a blocking queue to post `CompleteableFuture<Offset>` to the thread. The thread can then block on the queue, then drain it, query the head offset and complete all the drained futures.  Wrap all of this into a suitable class.

Reply:
Yup, I'll implement the blocking queue approach in a separate PR. For now, I've retargeted this PR to focus on having CLI requests wait until all previously-issued requests (from the particular CLI) have finished, which @apurvam pointed out has a simpler implementation.